### PR TITLE
SDCard updates: added BTT Octopus pro h723 v1.1

### PR DESCRIPTION
**Pull request by locki-cz**
**SDCard updates**

- Added definition for Octopus pro h723 v1 and v1.1 for sdcard update script.

Signed-off-by:  David Bucek <bucek.david@gmail.com>

### CODE REVIEW

**Comment by KevinOConnor** [comment]
Thanks.  I'm not sure if this PR is still relevant after #6301.  Maybe @dewhisna or @JamesH1978 knows.

-Kevin

**Comment by locki-cz** [comment]
SKR 3 and Octopus are different boards :)

**Comment by Arksine** [comment]
They are different, however the H723 variants appear to have identical configurations.  It might make sense to add an alias for the Octopus H723, however it isn't desirable to add duplicate board definitions.

**Comment by locki-cz** [comment]
Not possible create alias of board which isnt exist at that moment in latest klipper. :(

**Comment by Arksine** [comment]
It should be there, added in commit 20d87c73fe8750118741b8fc9f057c5861ee5df0.

File: scripts/spi_flash/board_defs.py [hunk 1]
```diff
@@ -159,6 +167,7 @@
     'btt-octopus-f446-v1.1': BOARD_DEFS['btt-octopus-f446-v1'],
     'btt-octopus-pro-f429-v1.0': BOARD_DEFS['btt-octopus-f429-v1'],
     'btt-octopus-pro-f446-v1.0': BOARD_DEFS['btt-octopus-f446-v1'],
+    'btt-octopus-pro-h723-v1.1': BOARD_DEFS['btt-octopus-pro-h723-v1'],
```
Comment (dewhisna) [inline scripts/spi_flash/board_defs.py:170]: @locki-cz -- I think all you need to do is first rebase your PR onto the current master and just add an alias for the Octopus Pro.  I did look at the schematic for it [https://raw.githubusercontent.com/bigtreetech/BIGTREETECH-OCTOPUS-Pro/master/Hardware/BIGTREETECH%20Octopus%20Pro%20V1.1-sch.pdf](https://raw.githubusercontent.com/bigtreetech/BIGTREETECH-OCTOPUS-Pro/master/Hardware/BIGTREETECH%20Octopus%20Pro%20V1.1-sch.pdf) and it appears to be the same pinout and MCU as the SKR3 for the SDIO connections.

So you should be able to change this line to

```
'btt-octopus-pro-h723-v1.1': BOARD_DEFS['btt-skr-3-h723']
```

after you rebase your PR.

You could also add an alias for the v1 entry as well:
```
'btt-octopus-pro-h723-v1': BOARD_DEFS['btt-skr-3-h723']
```
however, I don't see where BTT explicitly mentioned a v1 with the H723 -- I only see it on the v1.1 schematic.  But, it's possible they have a v1 with that MCU in addition to the v1.1.  Since most of the STM32 family has interchangeable parts, BTT has been making use of that lately -- probably because of supply-chain issues.

File: scripts/spi_flash/board_defs.py [hunk 2]
```diff
@@ -65,6 +65,14 @@
         'cs_pin': "PC11",
         'skip_verify': True
     },
+    'btt-octopus-pro-h723-v1': {
```
Comment (dewhisna) [inline scripts/spi_flash/board_defs.py:68]: @locki-cz  -- after you rebase your PR to the current master branch, you should be able to drop this extra entry and just add an alias for the octopus pro as shown below.

**Review by dewhisna — COMMENTED** [review]
@locki-cz -- if you rebase your PR to the current master branch, I think you can just edit your commit to alias the Octopus Pro to the SKR-3 entry and then just force push that to your branch on your fork to update your PR.  Even though the Octopus Pro and SKR-3 are different boards, the SDFlash details are identical.  See details in this review.

I can't just add another alias to my #6301 PR for the Octopus Pro, since it's already been merged, without opening another PR.  And since you have this PR open already, you can easily edit it to update it to just be an alias.

**Comment by dewhisna** [comment]
@locki-cz  -- How come you created a new PR (#6318) rather than just completing this one??  All you needed to do was rebase your `BTT-Octopus-pro-h723-v1.1` branch, tweak your commit, and force push it back to GitHub, with something like the following -- which assumes that the current working directory in the main `klipper` folder from your GitHub checkout and `origin` is the remote name for your GitHub fork (which is the default for git clone) and that you are currently on a checkout of your `BTT-Octopus-pro-h723-v1.1` branch with the single commit from your PR that needs to be rebased at the head:

```
git remote add upstream https://github.com/Klipper3d/klipper.git
git fetch upstream
git rebase upstream/master
```

Then edit the `board_defs.py` file to create the alias and then amend your existing commit (since it should be at the head of your `BTT-Octopus-pro-h723-v1.1` branch after the rebase)  with:

```
git add scripts/spi_flash/board_defs.py
git commit --amend
```

The `--amend` on the commit changes/updates the last commit rather than making another separate commit.

Then, assuming you've tested everything, force push it back to your GitHub branch, which would have automatically updated your PR using:

```
git push --force-with-lease origin BTT-Octopus-pro-h723-v1.1
```

And that's it...

**Comment by locki-cz** [comment]
I'm sorry, but I have no idea how to do it, a simpler solution is a new request. Github is still rocket science for me. And i dont want break klipper on my printer, simple way for me is clone latest klipper relase, edit file make from it own branch and request. Done, untouched printer :) No know any commands like you pro :)

**Comment by dewhisna** [comment]
@locki-cz -- I'm not sure I'd consider myself a "pro", but I do develop software for a living.  I will be the first to admit that git (the version control system, not necessarily GitHub itself) has a very steep learning curve compared to many other popular version control systems out there, like subversion, for example.  If you had pinged me, I would have been happy to walk you through it.

Heads-up though, I do see that your new PR was done on the `master` branch of your fork rather than in a separate branch.  That itself is OK, but if any other commit lands on the `master` branch of the main klipper repo before your PR does, or depending on how the klipper maintainers merge your PR, it will cause you a little grief the next time you either do anything on your fork on GitHub or try to do an update on your local repository clone, as the two repositories will have diverging paths.

Basically, you'll have to do a hard reset of your `master` branch in your local repository clone back to a point before that commit in order to do fast-forward updates again from the upstream, and then force-push that to your fork to get it resync'd.

So keep that in mind for the future.  No need to do anything about it at the moment though.
